src/code/z_skin_matrix.c

```diff
@@ -0,0 +1,804 @@
+#include <ultra64.h>
+#include <global.h>
+
+// clang-format off
+MtxF sMtxFClear = {
+    1.0f, 0.0f, 0.0f, 0.0f,
+    0.0f, 1.0f, 0.0f, 0.0f,
+    0.0f, 0.0f, 1.0f, 0.0f,
+    0.0f, 0.0f, 0.0f, 1.0f,
+};
+// clang-format on
+
+/**
+ * Multiplies a 4 component row vector [ src , 1 ] by the matrix mf and writes the resulting 4 components to xyzDest
+ * and wDest.
+ *
+ * \f[ [\texttt{xyzDest}, \texttt{wDest}] = [\texttt{src}, 1] \cdot [mf] \f]
+ */
+void SkinMatrix_Vec3fMtxFMultXYZW(MtxF* mf, Vec3f* src, Vec3f* xyzDest, f32* wDest) {
+    xyzDest->x = mf->wx + ((src->x * mf->xx) + (src->y * mf->yx) + (src->z * mf->zx));
+    xyzDest->y = mf->wy + ((src->x * mf->xy) + (src->y * mf->yy) + (src->z * mf->zy));
+    xyzDest->z = mf->wz + ((src->x * mf->xz) + (src->y * mf->yz) + (src->z * mf->zz));
+    *wDest = mf->ww + ((src->x * mf->xw) + (src->y * mf->yw) + (src->z * mf->zw));
+}
+
+/**
+ * Multiplies a 4 component row vector [ src , 1 ] by the matrix mf and writes the resulting xyz components to dest.
+ *
+ * \f[ [\texttt{dest}, -] = [\texttt{src}, 1] \cdot [mf] \f]
+ */
+void SkinMatrix_Vec3fMtxFMultXYZ(MtxF* mf, Vec3f* src, Vec3f* dest) {
+    f32 mx = mf->xx;
+    f32 my = mf->yx;
+    f32 mz = mf->zx;
+    f32 mw = mf->wx;
+
+    dest->x = mw + ((src->x * mx) + (src->y * my) + (src->z * mz));
```

Comment:
I suggest putting the dest->x without the extra \n next to the local setter, add a \n after dest->